main.cpp

```diff
@@ -7393,9 +7462,18 @@ static void check_otp_write_error(picoboot::command_failure &e, bool ecc) {
     }
 }
 
+bool settings_select_ecc(void) {
+    return settings.otp.ecc && !settings.otp.raw;
+}
+
+uint8_t otp_cmd_max_bits(void) {
+    return settings_select_ecc() ? 16 : 24;
+}
+
 typedef std::function<void(uint8_t *buffer, uint32_t len, picoboot_otp_cmd &otp_cmd)> otp_read_func_t;
 typedef std::function<void(uint8_t *buffer, uint32_t len, picoboot_otp_cmd &otp_cmd)> otp_write_func_t;
-void process_otp_json(json &otp_json, otp_read_func_t read_func, otp_write_func_t write_func) {
+void process_otp_json(json &otp_json, model_t model, otp_read_func_t read_func, otp_write_func_t write_func) {
+    int raw_max_bits = otp_cmd_max_bits();
```

Comment:
Technically this should always be the maximum, ignoring whether ECC is set, because the JSON file specifies ECC or not for each field so ignores the `settings.otp.ecc` and `settings.otp.raw` options.

Simple fix is
```suggestion
    int raw_max_bits = 24
```
but could add an `otp_raw_max_bits(void)` function?